packages/rum-core/src/common/apm-server.js

```diff
@@ -159,9 +162,26 @@ class ApmServer {
     if (environment) {
       configEndpoint += `&service.environment=${environment}`
     }
+
+    let localConfig = this._configService.getLocalConfig()
+    if (localConfig) {
+      configEndpoint += `&ifnonematch=${localConfig.etag}`
+    }
+
     return this._makeHttpRequest('GET', configEndpoint, { timeout: 5000 })
-      .then(response => {
-        return JSON.parse(response)
+      .then(xhr => {
+        const { status, responseText } = xhr
+        const etag = xhr.getResponseHeader('etag')
```

Comment:
shouldn't we account for multiple values if the header is set more than 1 time? Its an edge case, but pretty common if APM server has Load balancer in front.

Reply:
Can you please elaborate a bit on "Load balancer" scenario?